### PR TITLE
Get Bugzilla data for reportees

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: bugzilla_dashboard/query.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+class Query:
+    def __init__(self, name, params):
+        super().__init__()
+        self.name = name
+        self.params = params
+
+    def get_timeout(self):
+        return 600
```
Comment (La0) [inline bugzilla_dashboard/query.py]: WHy not a property here ?

File: bugzilla_dashboard/query.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        bugs = []
+        params = self.get_bz_params()
+
+        def bughandler(bug, data):
+            data.append(bug)
+
+        logger.info(f"Get bugs for {self.name}: starting...")
+        Bugzilla(
+            params, bughandler=bughandler, bugdata=bugs, timeout=self.get_timeout()
+        ).get_data().wait()
```
Comment (La0) [inline bugzilla_dashboard/query.py]: We need to rewrite libmozdata to use asyncio/await :smirk:

File: bugzilla_dashboard/query.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        if not out_dir:
+            return
+
+        os.makedirs(out_dir, exist_ok=True)
+
+        if compress:
+            data = json.dumps(data)
+            data = bytes(data, "utf-8")
+            data = gzip.compress(data, compresslevel=9)
+            with open(os.path.join(out_dir, file_name + ".gz"), "wb") as Out:
```
Comment (La0) [inline bugzilla_dashboard/query.py]: nit: use f strings here and below

File: bugzilla_dashboard/query.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                Out.write(data)
+        else:
+            with open(os.path.join(out_dir, file_name), "w") as Out:
+                json.dump(data, Out)
```
Comment (La0) [inline bugzilla_dashboard/query.py]: Nit: use `sort_keys=True, indent=4`

File: bugzilla_dashboard/query.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    @staticmethod
+    def get_args(description):
```
Comment (La0) [inline bugzilla_dashboard/query.py]: I would put that in a dedicated module `cli`

File: bugzilla_dashboard/query.py (hunk 4 above)
Comment (calixteman) [reply bugzilla_dashboard/query.py]: the file isn't consumed by humans... so I'd say it's useless.

**Comment by calixteman** [comment]
About the tests: yes we really need to have ones.

**Pull request merged**